[Mission] 1. 어서와 코딩은 처음이지? / Step2

wolframhwang opened the pull request:
@llghdud921 
제리 두 번재 PR 요청드립니다.

1. 미션에서 제시한 바와 같이, 함수를 두개로 나눠서 그림 그리는 책임을 위임했습니다.
2. 순서도는 함수와 for-in 루프를 컨테이너를 통해 표현했습니다.

감사합니다!

wolframhwang committed:
[feat] Step 2: 아이스크림 그리기
- 아이스크림, 막대기를 표현할 함수 두가지를 작성하였습니다.
- 각각, row, col 에 관하여 for 문을 통해 구현했습니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
[docs] Mission1: Step2: 아이스크림 그리기 순서도

아이스크림 그리는데 사용한 순서도 그림이 포함된 패치입니다.
함수와 포문을 컨테이너로 표현하였습니다

Signed-off-by: wolframhwang <hzw94@naver.com>

llghdud921 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,37 @@
+func makeIceCream() {
+    let height = 8
+    let width = 11
+    let rangeRow = 0..<height
```
`rangeRow`보다는 `rangeOfRow`가 더 명확한 네이밍이 될 수 있을 것 같네요 :)

llghdud921 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,37 @@
+    let rangeRow = 0..<height
+    let rangeCol = 0..<width
 
+    for _ in rangeRow {
```
조건문 너무 잘 작성해주셨어요👍👍

llghdud921 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,37 @@
 
-print("Hello, World!")
+func makeIceCream() {
+    let height = 8
```
`height`만으로는 어떤것의 `height`를 나타내는지 알 수 없는 것 같아요🥲 
좀 더 명확한 네이밍을 지어볼까요?

llghdud921 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,37 @@
+}
+
+func makeBar() {
+    let height = 5
```
전반적으로 하드코딩을 지양하고 변수로 설정하신 점 너무 칭찬드립니다👏

llghdud921 commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -8,5 +8,37 @@
+    let rangeCol = 0..<width
+
+    for _ in rangeRow {
+        for i in rangeCol {
```
[Swift API Design GuideLines](https://www.swift.org/documentation/api-design-guidelines/#use-terminology-well) 내 Avoid abbreviations을 보시면 나타나 있는데요!
Swift에서는 약어를 사용하는 것을 지양하고 있습니다:) `i`보다는 `index`라는 네이밍이 좋지 않을까 합니다

llghdud921 reviewed (changes requested):
안녕하세요!! 제리입니다~
Step2를 잘 진행해주셔서 인상깊었어요:)

### Good
- 함수 네이밍을 정말 잘해주셨습니다~! 💯 
- 함수의 기능분리가 잘 이루어져있어요 💯 
- 순서도에 코드가 아닌 문장형식으로 작성해주신 점 💯 
### 조언 드릴 부분
순서도를 작성해주셨는데요 일반적이 순서도와는 많이 달라보이네요🥲
[FlowChart](https://ko.wikipedia.org/wiki/%EC%88%9C%EC%84%9C%EB%8F%84) 위키백과에 나타나 있듯이 시작과 끝, 변수 선언, 처리를 화살표 방향으로 작성해주시면 좋을 것 같아요 :)

wolframhwang committed:
[refactor] CodeStarterCamp_Week1: Naming 리팩토링 사항 반영

- 범위 상수에 대해 명확한 네이밍이느 rangeOfRow / rangeOfCol 로 변경했습니다.
- 함수 콘텍스트에 존재하는 height / width를 콘텍스트 목적에 맞게 변경했습니다.
- 반복문에서 필요한 index 축약어 i를 index 로 변경했습니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

wolframhwang committed:
[docs] Mission1: Step2: 아이스크림 그리기 순서도 이미지 패치

아이스크림을 그리는데 사용한 순서도가 포함된 패치를 반영합니다.

Signed-off-by: wolframhwang <hzw94@naver.com>

llghdud921 reviewed (approved):
리팩토링, 순서도도 완벽하네요!!🙌  다음 스텝으로 넘어갑시다~!!!
